### ISSUE TITLE
Individual volume levels or mute other participants of the call

### ISSUE THREAD

**Issue by LuFlo** [nextcloud/spreed#3155]
**Is your feature request related to a problem? Please describe.**
It is very annoying when different participants have different volume levels set up and they are not able to fix this issue by their own.

**Describe the solution you'd like**
A way to alter the volume level for each participant (on client side), like in Discord or Teamspeak. A nice-to-have would be a mute button which decreases the volume down to zero. The volume control could be accessible by an icon next to the participants name in the sidebar list. On click or hover a volume slider appears, that way you don't need a right click action (mobile friendly).

**Describe alternatives you've considered**
Alter the volume each time another participant speaks. I think you get why this can get very annoying ;) Also I couldn't find any related issue. Note: I don't mean this to be a moderating option like in #282. Every participant should be able to setup their own volumes for each participant.

**Additional context**
/

**Comment by wagnerfl85** [comment]
I think this is a good idea. For me it would be great to have it like in Jitsi: You can mute and change volume level for a single person. For example when you have a meeting and one of the participants is in the same room like you and you hear him/her direct and in your headphone with a little time gap.

**Comment by LuFlo** [comment]
I was also thinking of Jitsi, but they say they are sending all voice channels to the client without mixing them. Can someone confirm if this is also the case with Nextcloud Talk? If not this could be kind of a headache to implement...

**Comment by nickvergessen** [comment]
Nextcloud Talk is peer-to-peer. So your browser directly sends the audio stream to the other participants browsers.

**Comment by LuFlo** [comment]
@nickvergessen thanks for the quick reply! That sounds good, so basically we "just" need some UI elements and the possibility to alter the volume of the streams.

**Comment by BloodyIron** [comment]
I really want the ability to adjust volume and mute per-person for myself, and not globally. Right now the mute function only mutes the person for EVERYONE, and you cannot unmute them once you've done it. It's really the opposite of what anyone would expect.

Also, if we could have the same slider amplify a specific person, that would be great too! Starts at the middle for everyone.

**Comment by DillFrescott** [comment]
Any progress on this? This is an essential feature for me.

**Comment by DillFrescott** [comment]
So I posted a $100 bounty for this on Fiverr (see linked image). If anyone here feels up to it I can just paypal you the money or another similar service.

https://i.imgur.com/pIN8GLZ.png

**Comment by BloodyIron** [comment]
Can we get this bumped up to a milestone instead of just backlog please?

**Comment by nickvergessen** [comment]
If anyone wants to pick it up they can even though it's backlog.
Backlog just means it's not going to be scheduled for the core team any time soon.

**Comment by DillFrescott** [comment]
@nickvergessen Would you accept payment to speed things up for this? Maybe we can arrange something...

**Comment by nickvergessen** [comment]
I can not and additionally it's not my area of expertise.

**Comment by DillFrescott** [comment]
Okay

**Comment by dillfrescott** [comment]
This really seems like something trivial to add, and so many new features are being added constantly it seems like this is intentionally being ignored.

I wouldnt say that if this was suggested last month, but its been 4 years...

**Comment by provokateurin** [comment]
@dillfrescott Feel free to put in the work to do it yourself, if you say it is so trivial to add. Your comment is not helpful to the conversation.

**Comment by dillfrescott** [comment]
There is no conversation. The last comment was in 2022.

**Comment by dillfrescott** [comment]
I suppose my original comment is a bit overly critical. I am unable to add this because i lack the programming capabilities, but I do know enough to know that volume bar's are not very difficult to add. Much more complex features are being added a lot. I contacted nick back in 2022 and offered funding to have it installed and it was still rejected. Again leading me to believe that the developers do not want this feature to be added.

**Comment by provokateurin** [comment]
Adding a volume bar itself in the UI might be easy, but wiring that up to control the actual volume is not as easy. You might be able to find someone to implement this at https://help.nextcloud.com/c/nextcloud-freelancing/48 where such requests are usually handled. The company and it's employees don't take "donations" for implementing features.

**Comment by dillfrescott** [comment]
A donation is a free contribution, if i am offering to pay for a service (being the addition of a volume bar) it wouldn't be considered a donation really at all.

**Comment by BloodyIron** [comment]
I've had to use a browser plugin "Disable Automatic Gain Control" just to stop this asinine UX behaviour. Frankly this is not something that users should even have to deal with (automatic volume adjustments in ways that they not only don't know is happening, but probably don't even want). My work-around is because I have not seen this topic been taken seriously, this is a UX problem and a very real one. I've had people not want to use NC Talk (Spreed) because of this single problem alone, because their mic suddenly is not as "good" as it was... just because WebRTC auto-attenuates without the user even being notified, and especially without the user having any control over it (on/off/adjustment,etc).

**Comment by dillfrescott** [comment]
I believe there was another thread (in nextcloud recognize) I made the comment to another user that the software was buggy and didnt work well, and the entire thread was locked and **then** I was scolded by an admin, not even allowing me the oppostunity to reply or refute.

Nextcloud is leaving a bad taste in my mouth, to say the least.

**Comment by provokateurin** [comment]
@dillfrescott please check the last message that was sent in the other issue by @GretaD.

As I said earlier already, please put in the required work yourself if it is important for you or if you are some kind of organization you might want to contact sales@nextcloud.com to get a contract.

**Comment by dillfrescott** [comment]
"Please put in the required work" isnt very customer friendly at all.

**Comment by provokateurin** [comment]
> customer friendly at all

I'm sorry, are you a customer of Nextcloud GmbH or one of it's partners? If so, please contact your Account manager. In any other case you are not a customer and not entitled to free work from anyone here.

I doubt any non-technical discussion here with you will lead to this issue getting solved, so please stop this behavior (and in other issues as well). Please read through https://nextcloud.com/contribute/code-of-conduct/ and adhere to it.

**Comment by dillfrescott** [comment]
Ok after looking over that, I dont see where I am being disrespectful of any other user, and it says nothing about not being able to admit when a component is in fact buggy or not working well. As that is all I did.

**Comment by dillfrescott** [comment]
Heres a tip: 

You dont even have to work on anything, but just saying: "We are sorry, we will look into this" is much nicer than just flat out saying "Do it yourself" and reflects better upon Nextcloud as a whole.

**Comment by provokateurin** [comment]
> I dont see where I am being disrespectful of any other user

You are constantly demanding that this problem should be fixed. That is not respectful to the people who develop this software. You're also blaming another developer working on a different part of Nextcloud while there is no relation between these two problems. I consider that not respectful and I doubt my colleague likes how you talk about him.

> it says nothing about not being able to admit when a component is in fact buggy or not working well

Nobody said this software is perfect and nobody said this issue is not real (I only subscribed to this issue because I once had this problem myself and wanted to be aware if/when it will be fixed in the future).
You are probably talking about the issue you linked, but as I said there is no connection between this and the other issue so you are only adding more blame and no helpful comments.

> You dont even have to work on anything, but just saying: "We are sorry, we will look into this" is much nicer than just flat out saying "Do it yourself" and reflects better upon Nextcloud as a whole.

That was exactly what was done in this issue, before you started demanding that it will be fixed. You deleted your initial comment, but saying that this is trivial to fix and not doing it yourself is literally just a demand that someone else should do it for you. You could have just said "I have this problem as well" (or even better just give the initial comment a thumbs up as that helps a lot more for visibility and doesn't add unnecessary comments) and be done with it, but instead you chose to blame other people and be disrespectful to them (and even other developers with no relation to this issue).

**Comment by BloodyIron** [comment]
Who are you talking to?

**Comment by provokateurin** [comment]
So they just deleted all their comments. I'll mark all of this discussion as offtopic because it is.

**Comment by BloodyIron** [comment]
So yeah, still hoping this can get improved upon here. It's really that big of a deal that it makes people not want to use the tool at all.

**Comment by dillfrescott** [comment]
I want to apologize.

I never should have complained. This is really a great project and I had absolutely no right trying to demand that a feature be implimented. This is a community project and not some big corporate program that I'm paying for or anything along those lines.

If I want a feature I will attempt to impliment it myself, somehow freelance it, or just wait patiently for it to be added.

I am a bit to passionate about Nextcloud as a whole, and I let that get the best of me.

**Comment by BloodyIron** [comment]
Well putting aside the social... uhhh... events... going on in this thread. I am looking forward to this becoming a thing! \o/

**Comment by Hadatko** [comment]
This would be great ;)

**Comment by Skyedra** [comment]
Would love to see this feature request get implemented one day as it's a must-have feature for community groups where everyone has their own varying hardware / mobile devices / etc.  It's a standard feature in the voice chat space versus discord, mumble, jitsi, etc.

Beyond that, I would also like to add two implementation considerations for anyone who ends up picking this up:

1)  Screen share audio should be separately controlled from voice audio to help solve the situation of a loud application audio but soft voice.

2)  Overgain should ideally be possible as some users have very quiet mics.  Example:  In a group of 6, it is a lot easier to overgain one person than quiet five others who otherwise are at reasonable volume by default.  (Mumble has a good implementation of this for reference)

Thanks for reading  :)

**Comment by trypr0x** [comment]
It's a pretty good feature to have, because you can host nextcloud on OCI, linode or digitalocean and just have a private "discord" with your friends that they dosen't have to rely on blocked network problems, they can just enter nextcloud talk and enter a call with your group of friends.

This is what I'm doing, and they prefer it, due to a few of my friends has their network blocked on discord, and some people on my friend group their parents do not let them use discord, [due to discord on brazil having a fame for killers, school shootings, raping and others](https://www.metropoles.com/sao-paulo/apos-mais-de-um-ano-discord-esconde-conversas-sobre-ataque-a-escola), This feature is great for self hosting nextcloud on OCI for example, and just have a "discord" for your own.

**Comment by ghost** [comment]
Im working from this problem. In the next week push result.

**Comment by ghost** [comment]
<img width="1160" height="1041" alt="Image" src="https://github.com/user-attachments/assets/25818d10-c3e8-48cf-9417-5741b4b81110" />

**Comment by Skyedra** [comment]
Amazing @ustyugov!  Does it work with screenshare also?  (If it is possible to control volume of screenshare independent from user, in case someone is playing loud game)

**Comment by G2G2G2G** [comment]
@nickvergessen unless you are going to work on it, the guy working on this deleted his account. So might as well remove from the milestones that keep getting pushed back for months
